src/hooks/useMedia.ts

```diff
@@ -0,0 +1,64 @@
+import { useEffect, useState } from "react";
+
+type UseMediaType = {
+    cameraList: InputDeviceInfo[];
+    micList: InputDeviceInfo[];
+    setCamera: React.Dispatch<React.SetStateAction<InputDeviceInfo | undefined>>;
```

Comment:
Don't expose this function directly, expose a wrapper function instead, so we can do some check before invoke the state change